p5rpc.modloader/Merging/Tbl/P3PTblMerger.cs

```diff
@@ -125,6 +130,65 @@ await Task.Run(async () =>
         });
     }
 
+    private async ValueTask PatchAnyFileInPak(RouteGroupTuple[] pakFiles, string tblPath, int ResolverSize, string[] cpks)
+    {
+        var route = tblPath.Substring(0, tblPath.LastIndexOf('\\'));
+        var tblName = Path.GetFileName(tblPath);
+        var candidates = FindInPaks(pakFiles, route, tblName);
+
+        if (candidates.Count == 0) return;
+
+        var extIndex = tblPath.IndexOf('.');
+        var dirIndex = tblPath.IndexOf('\\', extIndex);
+        var pathInCpk = '\\' + tblPath.Substring(0, dirIndex);
+        var pathInPak = tblPath.Substring(dirIndex + 1);
+
+        if (!_utils.TryFindFileInAnyCpk(pathInCpk, cpks, out var cpkPath, out var cpkEntry, out var fileIndex))
+        {
+            _logger.Warning("Unable to find TBL in any CPK {0}", tblPath);
+            return;
+        }
+
+        // Build cache key
+        string[] modIds = { "p5rpc.modloader" };
+        var cacheKey = MergedFileCache.CreateKey(tblPath, modIds);
+        var sources = candidates.Select(x => new CachedFileSource { LastWrite = File.GetLastWriteTime(x) }).ToArray();
+
+        if (_mergedFileCache.TryGet(cacheKey, sources, out var cachedFilePath))
+        {
+            _logger.Info("Loading Merged TBL {0} from Cache ({1})", tblPath, cachedFilePath);
+            _pakEmulator.AddFile(cachedFilePath, route, pathInPak);
+            return;
+        }
+
+        // Else Merge our Data
+        // First we extract.
+        await Task.Run(async () =>
+        {
+            _logger.Info("Merging {0} with key {1}.", tblPath, cacheKey);
+            await using var cpkStream =
+                new FileStream(cpkPath, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
+            using var reader = _criFsApi.GetCriFsLib().CreateCpkReader(cpkStream, false);
+            using var extractedPak = reader.ExtractFile(cpkEntry.Files[fileIndex].File);
+
+            var extractedTbl = _pakEmulator.GetEntry(new MemoryStream(extractedPak.RawArray), pathInPak);
+            if (extractedTbl == null)
+            {
+                _logger.Error($"Unable to extract {pathInPak} from {pathInCpk}");
+                return;
+            }
+
+            // Then we merge
+            byte[] patched;
```

Comment:
There's no reason to have this on two separate lines, just have `var patched = await PatchAny(extractedTbl.Value.ToArray(), candidates, ResolverSize);` 

I'm guessing this is copy pasted from one of the other similar functions so probably not your fault.

Reply:
If you change this, also do it in the P4G and P5R tbl mergers. Same with the following comment about patcher.Apply